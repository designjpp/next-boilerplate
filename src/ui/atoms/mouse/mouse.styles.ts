import { css } from '@emotion/css';
import xw from 'xwind';

const mouseClassNames = {
	base: xw`absolute top-0 left-0 z-10 pointer-events-none`,
	theme: xw`w-72px h-72px bg-mouse opacity-75 border-mouse border-3px rounded-full`,
	animation: xw`transition-all duration-100 ease-out motion-reduce:transform-none`,
};

export const mouseStyles = css`
	${mouseClassNames.base};
	${mouseClassNames.theme};
	${mouseClassNames.animation};
`;
