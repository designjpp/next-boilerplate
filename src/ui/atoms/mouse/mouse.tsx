import { useMousePosition } from 'lib/hooks';

import { mouseStyles } from './mouse.styles';

export const Mouse = ({ disabled = false }: { disabled?: boolean }) => {
	const { x, y } = useMousePosition();

	return (
		<span className='fixed top-0 bottom-0 left-0 right-0 z-10 h-screen pointer-events-none'>
			<span
				hidden={disabled}
				className={mouseStyles}
				style={{
					left: x - 72 / 2,
					top: y - 72 / 2,
				}}
			/>
		</span>
	);
};
