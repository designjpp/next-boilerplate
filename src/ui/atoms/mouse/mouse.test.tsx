import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import { create } from 'react-test-renderer';

import { Mouse } from './mouse';

describe('<Mouse>', () => {
	it('should match snapshot', async () => {
		const component = create(<Mouse />).toJSON();
		expect(component).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<Mouse />);
		const results = await axe(container);
		expect(results).toHaveNoViolations();
	});
});
