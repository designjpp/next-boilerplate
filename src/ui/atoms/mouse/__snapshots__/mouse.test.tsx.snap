// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mouse> should match snapshot 1`] = `
.emotion-0 {
  position: absolute;
  top: 0px;
  left: 0px;
  z-index: 10;
  pointer-events: none;
  --tw-bg-opacity: 1;
  width: 4.5rem;
  height: 4.5rem;
  background-color: rgba(18, 18, 18, var(--tw-bg-opacity));
  opacity: 0.75;
  border-color: rgba(255, 255, 255, 0.3);
  border-width: 0.1875rem;
  border-radius: 9999px;
  transition-property: all;
  transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
  transition-duration: 100ms;
}

@media (prefers-reduced-motion: reduce) {
  .emotion-0 {
    -webkit-transform: none;
    -moz-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

<span
  className="fixed top-0 bottom-0 left-0 right-0 z-10 h-screen pointer-events-none"
>
  <span
    className="emotion-0"
    hidden={false}
    style={
      Object {
        "left": -36,
        "top": -36,
      }
    }
  />
</span>
`;
